### PR TITLE
feat: Update asserts to use new winter23 Assert class - Async Recipes

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

Update asserts to use new winter23 Assert class. The new Assert class is really elegant compared to the older System methods.

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

**Comment by docbill** [comment]
Do people really find the new asserts somehow better?  Perhaps it is my bias from 20+ years of development using the original asserts, but this really seems like a case of salesforce breaking the foundational development rule of if it ain't broken, don't fix it.  If one does not know System.assert(...) is asserting if true, they really should not be a software devoloper.

File: force-app/tests/Async Apex Recipes/BatchApexRecipes_Tests.cls [hunk 1]
```diff
@@ -18,17 +18,17 @@ private inherited sharing class BatchApexRecipes_Tests {
         Test.stopTest();
 
         List<Account> checkAccounts = [SELECT Name FROM Account];
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Async Apex Recipes/BatchApexRecipes_Tests.cls:21]: Change to `areEqual`

File: force-app/tests/Async Apex Recipes/QueueableChainingRecipes_Tests.cls [hunk 2]
```diff
@@ -18,16 +18,16 @@ private inherited sharing class QueueableChainingRecipes_Tests {
         QueueableChainingRecipes.execute(null);
         Test.stopTest();
         List<Account> checkAccounts = [SELECT Description FROM Account];
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Async Apex Recipes/QueueableChainingRecipes_Tests.cls:21]: Change to `areEqual`

File: force-app/tests/Async Apex Recipes/QueueableRecipes_Tests.cls [hunk 3]
```diff
@@ -13,12 +13,12 @@ private inherited sharing class QueueableRecipes_Tests {
         Test.stopTest();
 
         List<Account> checkAccounts = [SELECT description FROM Account];
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Async Apex Recipes/QueueableRecipes_Tests.cls:16]: Change to `areEqual`

File: force-app/tests/Async Apex Recipes/QueueableWithCalloutRecipes_Tests.cls [hunk 4]
```diff
@@ -20,12 +20,12 @@ private inherited sharing class QueueableWithCalloutRecipes_Tests {
         Test.stopTest();
 
         List<Account> checkAccounts = [SELECT Description FROM Account];
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Async Apex Recipes/QueueableWithCalloutRecipes_Tests.cls:23]: Change to `areEqual`

File: force-app/tests/Async Apex Recipes/ScheduledApexDemo_Tests.cls [hunk 5]
```diff
@@ -7,7 +7,7 @@ private inherited sharing class ScheduledApexDemo_Tests {
         demoObj.runAtMidnight();
         Test.stopTest();
 
-        System.assertEquals(
+        System.Assert.areEqual(
```
Comment (pozil) [inline force-app/tests/Async Apex Recipes/ScheduledApexDemo_Tests.cls:10]: Fix param order

File: force-app/tests/Async Apex Recipes/ScheduledApexRecipes_Tests.cls [hunk 6]
```diff
@@ -19,19 +19,19 @@ private inherited sharing class ScheduledApexRecipes_Tests {
             WHERE id = :scheduledJobId
             LIMIT 1
         ];
-        System.assertEquals(
+        System.Assert.areEqual(
```
Comment (pozil) [inline force-app/tests/Async Apex Recipes/ScheduledApexRecipes_Tests.cls:22]: Fix param order in this test and the ones below

**Review by pozil — CHANGES_REQUESTED** [review]
Found some minor issues.

File: force-app/tests/Async Apex Recipes/BatchApexRecipes_Tests.cls (hunk 1 above)
Comment (msrivastav13) [reply force-app/tests/Async Apex Recipes/BatchApexRecipes_Tests.cls]: Fixed!

File: force-app/tests/Async Apex Recipes/QueueableChainingRecipes_Tests.cls (hunk 2 above)
Comment (msrivastav13) [reply force-app/tests/Async Apex Recipes/QueueableChainingRecipes_Tests.cls]: Fixed!

File: force-app/tests/Async Apex Recipes/QueueableRecipes_Tests.cls (hunk 3 above)
Comment (msrivastav13) [reply force-app/tests/Async Apex Recipes/QueueableRecipes_Tests.cls]: Fixed!

File: force-app/tests/Async Apex Recipes/ScheduledApexRecipes_Tests.cls (hunk 6 above)
Comment (msrivastav13) [reply force-app/tests/Async Apex Recipes/ScheduledApexRecipes_Tests.cls:22]: Fixed!

File: force-app/tests/Async Apex Recipes/ScheduledApexDemo_Tests.cls (hunk 5 above)
Comment (msrivastav13) [reply force-app/tests/Async Apex Recipes/ScheduledApexDemo_Tests.cls:10]: resolved!

**Review by pozil — APPROVED** [review]
All good, thanks for the update

**Pull request merged**